### ISSUE TITLE
Browers plugin? (eg. Firefox)

### ISSUE THREAD

**Issue by TobiasDev** [authpass/authpass#41]
I'm constantly struggling with a good KeePass plugin for Firefox. Tusk has a good idea, but it seems to never detect the proper fields. Is there plans to also develop a browser plugin for this one?

Thanks,
Tobias

**Comment by hpoul** [comment]
Do you mean for desktop firefox? No, not right now. Maybe providing an HTTP server so a plugin like https://addons.mozilla.org/en-US/firefox/addon/keepasshttp-connector/ could be used.. But I'm not sure how much better that would work than Tusk? Imho for desktop browsers a "Auto-Type"-like feature is the simplest thing most of the time..
Out of curiosity which platform are you on?

**Comment by TobiasDev** [comment]
I'm on Ubuntu and Windows 10.

**Comment by hpoul** [comment]
have you tried to get the AuthPass code to run on ubuntu/windows? I'm only on a mac and haven't got a chance to set up a VM..  it should be more or less just a matter of copying over the windows/linux scaffold from https://github.com/google/flutter-desktop-embedding/tree/master/example and running it..

**Comment by thirashima** [comment]
FWIW, Tusk is no longer maintained, as per their github repo.
https://github.com/subdavis/Tusk

**Comment by hpoul** [comment]
to be honest, for me the benefit/cost ratio to how much effort it would be to make a browser plugin to how much time it saves compared to just a native app with copy/paste (or auto type support) does not really motivate me.. if somebody would be interested to work on it, i'd be happy to help.. 
I still think the way to go would be to just make a browser plugin which communicates over some socket to the actual app, instead of the browser extension itself being the password manager.. probably just taking like the keepass http connector and implementing the server side in AuthPass would work as well..

**Comment by TobiasDev** [comment]
I have to agree with that. If you feel like it isn't worth adding this I think we can close the issue. :) There are alternatives.

**Comment by hpoul** [comment]
fwiw, it seems there is a new keepassxc browser extension which uses native communication. and it looks like the protocol is somehow documented: https://github.com/keepassxreboot/keepassxc-browser/blob/master/keepassxc-protocol.md so this might be worth checking out.. at least on macos and linux dart can easily bind on a named socket and use something like https://github.com/varjolintu/keepassxc-proxy-rust to proxy CLI native messages to the socket.. on windows probably a TCP port would be necessary, because dart does not yet implement windows named pipes nor named sockets on windows: https://github.com/dart-lang/sdk/issues/41161 .. but i don't really see how named sockets are much more secure than binding to a port on 127.0.0.1 anyway..

i'll reopen the issue for now.. it sounds reasonable, although i doubt i'll get to it in the near term.. but if anyone is interested giving it a go ;-)

**Comment by DavidJanko-gmail-com** [comment]
Hi,
just tried AuthPass and was looking for a way to connect it to other apps, especially browsers. Am I understanding correctly that there is no such thing yet?
In my opinion, the ideal password manager should serve as an authorization center for other applications, in the simplest case provide specific passwords in a safe way (I don't consider copy&paste to be safe) and manage and log this exchange centrally.
I'm going to try to install Flutter, an environment for Dart, and break down AuthPass from the sources. I am happy to participate in the development. I used to program in C, PHP on linux and windows, but I know absolutely nothing about Dart/Flutter or how browser plugins work. A TCP/IP server, on the other hand, should not be a problem in any language and any platform, perhaps even over SSL today. I don't have much time right now, but that could improve in the next six months.
Well, I'm going to see what I can do. If anyone would like to collaborate on this (perhaps just with advice on how such communication between applications could work or how Flutter works), let me know. I haven't programmed in a long time and I will have to learn GIT as well :).
I will follow this forum, or I am at david.janko@gmail.com, whatsApp with phone number +420 771 117 358.

**Comment by ecstrema** [comment]
I also think that a browser extension is the single thing preventing me from moving from keepassxc. 

It is just very very helpful to be able to fill up automatically passwords when opening a page and, and I think this is the most important point:

> I do not consider copying and pasting as safe. 

Keepassxc offers to delete the copied passwords from the clipboard after a few seconds, but ideally, it should have to go through the user, as users are inherently unsafe.

**Comment by ecstrema** [comment]
Without RTC needs though, what can be done under chromium is obtaining filesystem access rights for the single file: the passwords.kdbx file. Then the extension can do anything it wants with it, including autocomplete and all.

**Comment by Wallby** [comment]
I would like to see a unified AuthPass suite for mobile and PC. Mobile has this functionality built in, though not sure how well it works given that it doesn't always seem to use the correct fields. But I think this is what is really lacking. This project is the first to provide an actual decent UI on top of keepass, instead of the windows 98 one it has been using forever. But there is still some room for improvement to make this actually a tool that is nice to use.

Personally, I synchronize my .kdbx files using Microsoft's phone link. That works fine.

**Comment by Wallby** [comment]
> fwiw, it seems there is a new keepassxc browser extension which uses native communication

@hpoul I tried that, and it doesn't work all that well. Mainly because KeePass doesn't support multiple URLs per entry, and the browser extension doesn't let you manually select an option if it doesn't match the exact URL.

**Comment by Wallby** [comment]
Isn't this straightforward to turn into an extension? Given that there already is the browser version?

As to the only missing feature, detecting form fields, there are plenty of open source projects that could be used as an example?
https://github.com/mozilla/form-fill-examples
